src/hooks/requests/useAuditLogs.ts

```diff
@@ -0,0 +1,61 @@
+import { useMutation, useQuery } from '@tanstack/react-query'
+import { apiClient } from 'api'
+import { endpoints } from 'api/endpoints'
+import { downloadFile } from 'helpers'
+import useFilters from 'hooks/useFilters'
+import i18n from 'i18n/i18n'
+import { isEmpty, map, omit } from 'lodash'
+import {
+  AuditLogPayloadType,
+  AuditLogsResponseDataType,
+  EventTypes,
+} from 'types/auditLogs'
+
+export const useFetchAuditLogs = () => {
+  const { filters, handleFilterChange, handlePaginationChange } =
+    useFilters<AuditLogPayloadType>()
+
+  const { isLoading, data } = useQuery<AuditLogsResponseDataType>({
+    enabled: !isEmpty(omit(filters, ['page', 'per_page'])),
+    queryKey: ['auditLogs', filters],
+    queryFn: () => apiClient.get(endpoints.AUDIT_LOGS, filters),
+    keepPreviousData: true,
+  })
+  const { meta: paginationData, data: logsData } = data || {}
+
+  return {
+    isLoading,
+    logsData,
+    paginationData,
+    handleFilterChange,
+    filters,
+    handlePaginationChange,
+  }
+}
+
+export const useExportAuditLogsCSV = () => {
+  const { mutateAsync: exportCSV, isLoading } = useMutation({
+    mutationKey: ['csv'],
+    mutationFn: async (payload: AuditLogPayloadType) =>
+      apiClient.get(endpoints.EXPORT_AUDIT_LOGS, payload),
+    onSuccess: (data) => {
+      downloadFile({
+        data,
+        fileName: 'audit_logs.csv',
+      })
+    },
+  })
+  return {
+    isLoading,
+    exportCSV,
+  }
+}
+
+export const useEventTypesFetch = () => {
```

Comment:
**[blocker]**

Should be under hooks and renamed, since this is not fetching anything and not related to API requests at all. Personally I wouldn't even create a helper for it, but rather map it where it's needed 

      options: map(EventTypes, (type) => ({
        value: type,
        label: i18n.t(`logs.event_type.${type}`),
      })),
      
But if you want to create a hook out of it, then it should just be under /hooks folder

Reply:
Extra note: avoid importing i18n directly and using it like this, except for cases, when there is no other option. the "t" function should always be used from the hook, to make sure the correct translations are taken, when the pages language is changed. importing i18n directly and using the "t" function from there won't update right away and will cause issues. Once we add actual language switching we will need to go over the places where it is imported directly and make sure that they will refresh correctly, so we want to keep those places to a minimum